website/src/main/java/com/google/step/data/OrganizationInfo.java

```diff
@@ -0,0 +1,61 @@
+package com.google.step.data;
+
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Query;
+import java.util.Objects;
+import javax.servlet.http.HttpServletRequest;
+import java.util.ArrayList;
+import org.apache.commons.validator.UrlValidator;
+
+public final class OrganizationInfo {
+  private final Entity entity;
+
+  public OrganizationInfo(HttpServletRequest request) {
+    this.entity = getOrgEntityFrom(request);
```

Comment:
In general, you want to avoid doing "work" in constructors.

What exactly is the behavior of `request.getParameter` if the request doesn't have the parameter? Based on the code below it seems like you'd get the empty string? What about type mismatches?

This also somewhat complicates testing. The merge logic doesn't depend on the entities coming from a HttpServletRequest, but there's no way to construct an OrganizationInfo without one.

It might make more sense to have a constructor which takes in an Entity to allow for easier testing.

Also, please add tests.

Reply:
I fixed the first part using a static function kinda like how you told me to do back in the portfolio for comments. 

And I definitely forgot about testing... Peng would be disappointing. But I'll get started on that. 

Also request.getParameter returns null if the parameter doesn't exist and a string if it does.